lib/src/native/realm_core.dart

```diff
@@ -664,6 +657,61 @@ class _RealmCore {
     return RealmHandle._(realmPtr);
   }
 
+  RealmAsyncOpenTaskHandle createRealmAsyncOpenTask(FlexibleSyncConfiguration config) {
+    final configHandle = _createConfig(config);
+    final asyncOpenTaskPtr =
+        _realmLib.invokeGetPointer(() => _realmLib.realm_open_synchronized(configHandle._pointer), "Error opening realm at path ${config.path}");
+    return RealmAsyncOpenTaskHandle._(asyncOpenTaskPtr);
+  }
+
+  Future<RealmHandle> openRealmAsync(RealmAsyncOpenTaskHandle handle, CancellationToken? cancellationToken) {
+    final completer = CancellableCompleter<RealmHandle>(cancellationToken);
+    final callback =
+        Pointer.fromFunction<Void Function(Handle, Pointer<realm_thread_safe_reference> realm, Pointer<realm_async_error_t> error)>(_openRealmAsyncCallback);
+    final userData = _realmLib.realm_dart_userdata_async_new(completer, callback.cast(), scheduler.handle._pointer);
+    _realmLib.realm_async_open_task_start(
+      handle._pointer,
+      _realmLib.addresses.realm_dart_async_open_task_callback,
+      userData.cast(),
+      _realmLib.addresses.realm_dart_userdata_async_free,
+    );
+
+    return completer.future;
+  }
+
+  static void _openRealmAsyncCallback(Object userData, Pointer<realm_thread_safe_reference> realmSafePtr, Pointer<realm_async_error_t> error) {
+    return using((Arena arena) {
+      final completer = userData as Completer<RealmHandle>;
```

Comment:
Yes, a `CancellableCompleter<T>` is also a `Completer<T>` .. was about to quip a bit 😄